docs/userman/gui/extension/chart.md

```diff
@@ -0,0 +1,184 @@
+    const dashboardLayout = useDynamicJsonProperty(props.layout, props.defaultLayout || "", {} as Partial<Layout>);
+
+    const data = useMemo(() => {
+        if (Array.isArray(value)) {
+            return value as Data[];
+        }
+        return [] as Data[];
+    }, [value]);
+
+    const baseLayout = useMemo(() => {
+        const layout = {
+            ...dashboardLayout,
+        };
+        return layout as Partial<Layout>;
+    }, [dashboardLayout]);
+
+    return (
+        <div>
+            <Plot data={data} layout={baseLayout} />
+        </div>
+    );
+};
+
+export default Dashboard;
+```
+
+The detailed explanation of the code is as follows:
+
+- The **Dashboard** component accepts two props: *data* and *layout*.
+- The [`useDynamicJsonProperty()`](../../../refmans/reference_guiext/functions/useDynamicJsonProperty.md) hook is used to handle dynamic properties. It accepts the following arguments:
+
+    - *data*: The dynamic property that holds the data for the chart.
+    - *defaultData*: The default value for the data property.
+    - *layout*: The dynamic property that holds the layout properties for the chart.
+    - *defaultLayout*: The default value for the layout property.
+
+- The **Plot** component utilizes the [`react-plotly.js`](https://github.com/plotly/react-plotly.js) library to render the chart. It accepts two props:
+
+    - *data*: An array of data objects to define the chart's datasets.
+    - *layout*: An object specifying the layout properties of the chart.
+
+For more information on how to use `react-plotly.js`, refer to the official documentation [here](https://github.com/plotly/react-plotly.js).
+
+
+
+## Exporting the React component {data-source="gui:doc/extension/example_library/front-end/src/index.ts"}
+
+When the component is entirely defined, it must be exported by the library's JavaScript bundle.
+This is done by adding the export directive in the file *<project dir>/<package dir>front-end/src/index.ts*.
+
+```ts title="index.ts"
+import Dashboard from "./Dashboard";
+
+export { Dashboard };
+```
+
+## Configuring the webpack configuration {data-source="gui:doc/extension/example_library/front-end/webpack.config.js#L39"}
+
+To bundle the React component, you must configure the webpack configuration file.
+Since Plotly.js is a JavaScript library, you need to include JavaScript in the webpack’s resolve.extensions setting.
+
+```js title="webpack.config.js"
+module.exports = {
+    // ...
+    resolve: {
+        extensions: [".ts", ".tsx", ".js"],
+    },
+    // ...
+};
+```
+
+After configuring the webpack file, you can build the library as mentioned in the [Building the front-end module](dynamic_element/index.md) section.
+
+## Using the element {data-source="gui:doc/extension/example_library/dashboard.py"}
+
+To use the *dashboard* element in a Python script, you can follow the example below:
+
+```python title="dashboard.py"
+trace1 = {
```

Comment:
Please rename 'trace' to 'set', to align with the name.
The point is: at the dataset level, we're not talking about traces. This happens only when displayed.